GCDWebServer/Core/GCDWebServerConnection.m

```diff
@@ -278,7 +278,7 @@ - (void)_writeHeadersWithCompletionBlock:(WriteHeadersCompletionBlock)block {
 
 - (void)_writeBodyWithCompletionBlock:(WriteBodyCompletionBlock)block {
   GWS_DCHECK([_response hasBody]);
-  [_response performReadDataWithCompletion:^(NSData* data, NSError* error) {
+  void (^readDataBlock)(NSData*, NSError*) = ^(NSData* data, NSError* error) {
```

Comment:
Instead of patching the call site, which is fragile, patch instead where GCDWebServer calls back to the app at `-performReadDataWithCompletion`: try replacing

    [_reader asyncReadDataWithCompletion:block];

by

    [_reader asyncReadDataWithCompletion:[block copy]];